pallets/fiat_filter/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+            Some(did::Call::remove(_to_remove, _sig)) => return Ok(PRICE_DID_OP),
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(anchor::Call::deploy(bytes)) => {
+                return Ok(PRICE_ANCHOR_OP_PER_BYTE * bytes.len() as u64)
+            }
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(blob::Call::new(blob, _sig)) => {
+                let size: u64 = 64 + blob.blob.len() as u64;
+                return Ok(PRICE_BLOB_OP_PER_BYTE * size);
+            }
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(revoke::Call::new_registry(_id, _registry)) => {
+                return Ok(PRICE_REVOKE_REGISTRY_OP)
+            }
+            Some(revoke::Call::remove_registry(_rm, _proof)) => {
+                return Ok(PRICE_REVOKE_REGISTRY_OP)
+            }
+
+            Some(revoke::Call::revoke(revocation, _proof)) => {
+                // 32 bytes for the registry_ID, 32 bytes per revocation
+                let size: u64 = 32 + 32 * revocation.revoke_ids.len() as u64;
+                return Ok(PRICE_REVOKE_REVOCATION_OP * size);
+            }
+            Some(revoke::Call::unrevoke(unrevoke, _proof)) => {
+                // 32 bytes for the registry_ID, 32 bytes per unrevocation
+                let size: u64 = 32 + 32 * unrevoke.revoke_ids.len() as u64;
+                return Ok(PRICE_REVOKE_REVOCATION_OP * size);
+            }
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(attest::Call::set_claim(_attester, attestation, _sig)) => {
+                // 8 bytes for the priority, plus IRI bytes
+                let size: u64 = 8 + attestation.iri.as_ref().unwrap_or(&[1].to_vec()).len() as u64;
+                return Ok(PRICE_ATTEST_OP_PER_BYTE * size);
+            }
+            _ => {}
+        }
+
+        // return error if Call not in list
+        // uses Pays::Yes to prevent spam
+        return Err(Error::<T>::UnexpectedCall.into());
+    }
+
+    fn compute_call_fee_dock_(
+        call: &<T as Config>::Call,
+    ) -> Result<BalanceOf<T>, DispatchErrorWithPostInfo> {
+        // the fee in fiat is expressed in nUSD (1 billionth USD)
+        let fee_usd_billionth: AmountUsd = Self::get_call_fee_fiat_(call)?;
+
+        // the DOCK/USD rate in the price_feed pallet is the price of 1DOCK,
+        // expressed in USD_1000th/DOCK (as u32) (== USD/1000DOCK)
+        let dock_usd1000th_rate: u32 =
+            match <T as Config>::PriceProvider::optimized_get_dock_usd_price() {
+                Some((rate, _weight)) => rate,
+                None => return Err(Error::<T>::NoPriceFound.into()),
+            };
+
+        // we want the result fee, expressed in µDOCK (1 millionth DOCK)
+        // no need for any multiplication factor since (1/1B USD)/(1/1000 USD/DOCK) already == (1/1M DOCK)
+        let fee_microdock: u64 = fee_usd_billionth
```

Comment:
Can you introduce a fallback constant like MinPrice or something here to be used in case `dock_usd1000th_rate` is 0 or there is an error fetching price through `optimized_get_dock_usd_price`.